### ISSUE TITLE
Unexpected "fraction wildcard might be negative" error with QPs

### ISSUE THREAD

**Issue by viper-admin** [viperproject/silicon#409]
> Created by **@mschwerhoff** on 2020-02-14 14:22

The following snippet verifies in Carbon, but not in Silicon:

```plaintext
field f: Int

method test0(xs: Set[Ref], x: Ref) {
  inhale x in xs ==> acc(x.f, wildcard)
  exhale x in xs ==> acc(x.f, wildcard)
  exhale x in xs ==> acc(x.f, wildcard)
}

method test01(xs: Set[Ref]) {
  inhale forall x: Ref :: {x.f} x in xs ==> acc(x.f, wildcard)
  exhale forall x: Ref :: {x.f} x in xs ==> acc(x.f, wildcard)
  exhale forall x: Ref :: {x.f} x in xs ==> acc(x.f, wildcard) // Unexpectedly fails in Silicon
}
```

‌

**Comment by mschwerhoff** [comment]
Sorry, too quick, referenced change in not-yet-merged branch

**Comment by mschwerhoff** [comment]
Resolving commit d630db7 was merged in as part of pull request #497
